oktoberfest/preprocessing/preprocessing.py

```diff
@@ -173,15 +179,29 @@ def filter_peptides(peptides: pd.DataFrame, min_length: int, max_length: int, ma
 
     :return: The filtered dataframe given the provided constraints.
     """
-    return peptides[
-        (peptides["PEPTIDE_LENGTH"] <= max_length)
-        & (peptides["PEPTIDE_LENGTH"] >= min_length)
-        & (peptides["PRECURSOR_CHARGE"] <= max_charge)
-        & (~peptides["MODIFIED_SEQUENCE"].str.contains(r"\(ac\)"))
-        & (~peptides["MODIFIED_SEQUENCE"].str.contains(r"\(Acetyl \(Protein N-term\)\)"))
-        & (~peptides["MODIFIED_SEQUENCE"].str.contains(r"\[UNIMOD\:21\]"))
-        & (~peptides["SEQUENCE"].str.contains("U|X"))
-    ]
+    if isinstance(peptides, pd.DataFrame):
+        return peptides[
+            (peptides["PEPTIDE_LENGTH"] <= max_length)
+            & (peptides["PEPTIDE_LENGTH"] >= min_length)
+            & (peptides["PRECURSOR_CHARGE"] <= max_charge)
+            & (~peptides["MODIFIED_SEQUENCE"].str.contains(r"\(ac\)"))
+            & (~peptides["MODIFIED_SEQUENCE"].str.contains(r"\(Acetyl \(Protein N-term\)\)"))
+            & (~peptides["MODIFIED_SEQUENCE"].str.contains(r"\[UNIMOD\:21\]"))
+            & (~peptides["SEQUENCE"].str.contains("U|X"))
+        ]
+    elif isinstance(peptides, AnnData):
+        selection = peptides.obs[
+            (peptides.obs["PEPTIDE_LENGTH"] <= max_length)
```

Comment:
lots of code duplication. Better: have the ifelse only to define the object you wanna filter, then write the filters only for that object once after the if else.